### ISSUE TITLE
whatshap stats - definition of phased variants

### ISSUE THREAD

**Issue by pontushojer** [whatshap/whatshap#414]
I am a bit confused on this after looking through the code and documentation. 

This is how the number of phased variants (parameter/column `phased` in output) is currently calculated.
https://github.com/whatshap/whatshap/blob/ee79863c30b59a9031812a99ea80b7666e9342cb/whatshap/cli/stats.py#L304

i.e. its the number of phased variants in non-singleton blocks. This also means that it will be that same values as for `variant_per_block_sum` 

https://github.com/whatshap/whatshap/blob/ee79863c30b59a9031812a99ea80b7666e9342cb/whatshap/cli/stats.py#L312

which makes this statistic a bit redundant.

It makes sense in a way that blocks of length one are not considered phased in these statistics. However in the documentation `phased` is currently defined as: 
> The number of biallelic, heterozygous variants that are marked as phased in the input VCF. This is again a subset of heterozygous_variants. Also, phased + unphased + singletons = heterozygous_variants.

The first sentence would include singletons if they are still marked phased. But then the statment "phased + unphased + singletons = heterozygous_variants" is no longer true. 

I also noted a statement below this stating "singletons + variant_per_block_sum = phased". This is currently the same as "singletons + phased = phased" which is only true if singletons are zero.  

**TLDR** What definition for `phased` (the number of phased variants) should we have in the statistics? I think it should be as now, i.e. excluding singletons, in which case the documentation should be updated to clarify this.

**Comment by marcelm** [comment]
Thanks, good observations - I’m glad someone finally took the time to double-check what I wrote!

> which makes this statistic a bit redundant.

Yes, this is surely an oversight. It was already in the very first commit that added the `stats` command, so nothing that has been added since then. We should document this.

> It makes sense in a way that blocks of length one are not considered phased in these statistics.

I think this is mainly there for dealing with artifacts. Perhaps a phased VCF was filtered, removing all but one of the variants in a block (this could even happen when running `stats` with `--only-snvs`). And in principle, a tool should not be able to get a high phased percentage by outputting only singleton blocks.

Treating singletons as phased would make sense if the order of the haplotypes has a meaning (haplotype 1=mother, haplotype 2=father or so), but then the user can just add up the `phased` and `singleton` numbers themselves.

So I agree that `phased` should continue to not include singletons.

I wrote the descriptions in the documentation by looking at the code. If there’s a discrepancy, it should be fixed in the documentation.

> I also noted a statement below this stating "singletons + variant_per_block_sum = phased". This is currently the same as "singletons + phased = phased" which is only true if singletons are zero.

I suggest to remove that statement entirely. The only thing mentioning `variant_per_block_sum` should be a sentence explaining that it is the same as `phased`.

**Comment by pontushojer** [comment]
Thanks for clarifying this!  I agree with everything here. Opened a PR to update the docs.
